### DISCUSSION TITLE
Is it possible to have data reloaded when a the meta data (coming from the URL route parameters) change?

### DISCUSSION THREAD

**Discussion by TomCaserta** [refinedev/refine#6318] [Q&A] [answered]
Essentially I'm reading over the multitenancy documentation https://refine.dev/docs/guides-concepts/multi-tenancy/ and I've implemented what it suggests:

1. Route parameter called `:tenantId`
2. My data provider uses the `tenantId` found in the meta and sends it to my endpoints via a header

The issue seems to be when I replace the tenantId parameter in the URL it doesn't actually reload the data. 

After looking at the examples it seems it uses the params and the tenant ID as a filter parameter in the multi tenancy strapi example.

This seems to be different to the suggested approach and of course works in that scenario as the param changing causes the query key to change as the filters are changing. As far as I can tell the meta (combinedMeta) is not taken into consideration in the `useList` query key creation, is this intentional?:

https://github.com/refinedev/refine/blob/master/packages/core/src/hooks/data/useList.ts#L238

Is there any way to make this work using route parameters and without specifying a filter on every usage of a useList/useOne/useMany hook?

**Comment by aliemir** [comment, marked as answer]
Hey @TomCaserta sorry for the late answer. You're right, query keys should have used the `combinedMeta`. This case was overlooked in the tests and the implementation ended up using the wrong composition of the `meta` variables. I'll convert this to an issue. I'm hoping to get a fix implemented and released within our next release cycle 🚀
